### PR TITLE
Spec 0004 community wording suggestions

### PR DESCRIPTION
**Pull request by rossbar**
A couple minor wording suggestions based on discussions from the NumPy triage meeting on June 14th, 2023.

The goal of the proposed rewording is to help clarify the underlying motivation for the admin selection process, and to clarify the distinction between the adminstrators' decision-making  from that of the individual projects nominating representatives.

I also proposed to remove a sentence in d0341ca which had caused some confusion. In practice, this would require each project to know the complete set of existing administrators and their affiliations which seems overly burdensome (especially as the number of projects grows). My interpretation of the original intent was to make clear that the pool of administrators should be representative of the whole community, and to ensure that any one organization is not over-represented. My hope is that the proposed rewording emphasizes this point but in a slightly less prescriptive manner.

### CODE REVIEW

**Comment by tupui** [comment]
Yay, thanks for doing that! Would that close #223?

File: spec-0004/index.md
Comment (tupui) [inline spec-0004/index.md:101]: Maybe we should move this after the next paragraph. This part is linked to the project, not the org itself.

**Review by tupui — COMMENTED** [review]
Rewording reads well to me (and, to my knowledge, corresponds to the intent we collectedly had when we wrote that) 👍 I left a suggestion

File: spec-0004/index.md
Comment (stefanv) [inline spec-0004/index.md:106]:
```suggestion
Adding new administrators requires at least an issue to be opened.
```

File: spec-0004/index.md
Comment (stefanv) [inline spec-0004/index.md:118]:
```suggestion
A project's chosen administrators should each create an
account on https://anaconda.org and share their usernames with the
project.
```

With the project, or with... the nightly wheel admins? And how?

File: spec-0004/index.md
Comment (stefanv) [inline spec-0004/index.md:121]: Perhaps move this to before the previous paragraph?

```suggestion
To increase resilience, we suggest that each project have at least two administrators.
```

I think (a) any chosen representative is, err, representative of the underlying community. And I'm not sure what it would mean to choose two people broadly; unless it means not two people sitting in the same office, or working in the same company. We could be more explicit about that then:


```suggestion
To increase resilience, we suggest that each project have at least two administrators who are not physically co-located.
```

**Review by stefanv — APPROVED** [review]

File: spec-0004/index.md
Comment (mattip) [inline spec-0004/index.md:100]:
```suggestion
The site admins are drawn from active members from the scientific Python community.
```

File: spec-0004/index.md
Comment (mattip) [inline spec-0004/index.md:108]:
```suggestion
requesting access to upload wheels a admin has to respond to that request.
```

File: spec-0004/index.md
Comment (mattip) [inline spec-0004/index.md:115]:
```suggestion
A project's chosen representative should each create an
```

File: spec-0004/index.md
Comment (mattip) [inline spec-0004/index.md:118]:
```suggestion
To increase resilience, we suggest that each project have at least two registered
representatives.
```

File: spec-0004/index.md
Comment (mattip) [inline spec-0004/index.md:120]:
```suggestion
The representative can then generate a personal access token at
```

File: spec-0004/index.md
Comment (mattip) [inline spec-0004/index.md:121]:
```suggestion
https://anaconda.org/[user]/settings/access and use it in CI to upload
wheels.
```

File: spec-0004/index.md
Comment (mattip) [inline spec-0004/index.md:111]:
```suggestion
amount of due dilligence before giving people access. This is because once a project is given
```

**Review by mattip — COMMENTED** [review]
I suggest to clearly deliniate between admins, projects, and representatives. This was my original confusion with the section.

File: spec-0004/index.md
Comment (mattip) [reply spec-0004/index.md:108]: Whoops, I missed "After someone creates" -> "After a project creates"

File: spec-0004/index.md
Comment (mattip) [reply spec-0004/index.md:118]: Hmm "and share their usernames with the project" is unclear. What is the intent?

File: spec-0004/index.md
Comment (mattip) [reply spec-0004/index.md:120]: A project's chosen representatives

File: spec-0004/index.md
Comment (bsipocz) [inline spec-0004/index.md:107]: from the review above
```suggestion
After a project creates an issue on https://github.com/scientific-python/upload-nightly-action
```

File: spec-0004/index.md
Comment (mattip) [inline spec-0004/index.md:117]: Thanks. This bit remains unclear: with whom are the representatives supposed to share their usernames?

**Comment by mattip** [comment]
I guess it is easy to get the approval of the projects that have endorsed the SPEC for these changes: they are networkx (@rossbar proposed the PR) and scikit-image (@stefanv approved the PR).

**Review by jarrodmillman — APPROVED** [review]

**Comment by jarrodmillman** [comment]
@mattip Yes, getting approval from the endorsing projects is trivial. Why don't you make sure the wording for the SPEC is okay for you. Once you are happy with the text and process just let us know and I will be sure to ask the scikit-image and NetworkX teams to re-endorse your changes or whatever would make you feel most comfortable. Thanks for taking such a close look at the wording of this document.

File: spec-0004/index.md
Comment (jarrodmillman) [reply spec-0004/index.md:117]: We could create a private mailing list for just this purpose. It suspect this process won't be used more than a few times over the next few years, so maybe a mailing list is overkill. We could also just close the list to new projects until we get more space and sort out how to handle communications. @mattip What would you prefer?

File: spec-0004/index.md
Comment (jarrodmillman) [inline spec-0004/index.md:115]:
```suggestion
A project's chosen representatives should each create an
```

File: spec-0004/index.md
Comment (mattip) [reply spec-0004/index.md:117]: I don't think the SPEC needs to dive into the mechanism. My comment is about the last word "project" in the text. Should it be changed to "site admins"? Or is it something else?

Comment (stefanv) [reply spec-0004/index.md:117]: I had the same question. I think site admins or similar is correct.

File: spec-0004/index.md
Comment (mattip) [inline spec-0004/index.md:117]: I think this is the intention?
```suggestion
site admins.
```

Comment (rossbar) [reply spec-0004/index.md:117]: I actually read it differently: i.e. it is recommended that the representatives share the account names with the projects so that other members of the project are aware of the account(s) for uploading wheels.

Obviously it's still not very clear :upside_down_face:

Comment (mattip) [reply spec-0004/index.md:117]: I would prefer that the manner and methods in which projects internally share information remain outside the scope of this SPEC. If indeed that is the authors' intent, then I would propose removing the "share their username" clause.

Comment (tupui) [reply spec-0004/index.md:117]: So this is what I wanted to say.
```suggestion
admins of the Scientific Python organization on Anaconda.
```

For sharing the info within your own project, this is your call on how you administrate things. Here we just need to know who to give access to.

**Comment by tupui** [comment]
@mattip @rossbar does the current document answer your questions? Could you either approve or make appropriate suggestions?

**Comment by rossbar** [comment]
I'm not able to approve my own PR, but the wording LGTM. No doubt it can be made more clear; please don't hesitate to push to this PR as needed!

**Comment by mattip** [comment]
LGTM thanks.

**Review by tupui — APPROVED** [review]
Great then 😃 thank you for the discussions and rephrasing!

**Pull request merged**
